internal/check/check.go

```diff
@@ -0,0 +1,153 @@
+package check
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"strconv"
+	"sync"
+	"text/tabwriter"
+	"time"
+
+	"golang.org/x/sync/semaphore"
+)
+
+type RunFunc func(ctx context.Context) (string, error)
```

Comment:
In general, I think the output will need to be some sort of structured object instead of just a `string`. This is fine for now though.